clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1363,17 +1365,18 @@ event_pre_syscall(void *drcontext, int sysnum)
 
 #ifdef BUILD_PT_TRACER
     if (op_offline.get_value() && op_enable_kernel_tracing.get_value()) {
-        if (!syscall_pt_trace_t::is_syscall_pt_trace_enabled(sysnum)) {
```

Comment:
Why did we need to move this? If `!is_syscall_pt_trace_enabled`, then does the  `data->syscall_pt_trace.get_cur_recording_sysnum() != INVALID_SYSNUM` check even matter?

Reply:
Yes. When encountering a new system call, verifying whether the tracing of the previous system call has been stopped is necessary. Regardless of whether or not the new system call can be traced, the checking of the previous system call should not be omitted. Skip the checking may result in the recording of more noise instructions.